networkx/linalg/algebraicconnectivity.py

```diff
@@ -244,7 +244,7 @@ def P(x):
             W -= (W.T * X * X.T).T
             project(W)
             # Compute the diagonal of P * L * P as a Jacobi preconditioner.
-            D = L.diagonal()
+            D = L.diagonal().astype(float)
             D += 2. * (asarray(X) * asarray(W)).sum(axis=1)
```

Comment:
Good...  I'm just getting global/local binding confused with the way Matlab does it.

Still having trouble understanding the suggested way to "reinitialize the solver".  Does that mean make it
```solver.solve(W, 0.1, D)``` instead of ```solver.solve(W, 0.1)``` and inside _PCG_Solver._solve we use ```D * r``` instead of ```M(r)```?  If so, I think that would help readability.

Meanwhile @ysitu what do you think about ```.astype(float)``` vs replacing the ```+=``` lines with a single assignment statement? I'm ready to go with whatever you like here.